### PR TITLE
[PROCS-3633] Stop populating unused fields in CPUInfo in the agent payload

### PR DESCRIPTION
**Pull request by levin0710**
### What does this PR do?

Stops populating the unused fields so we can save on the size of our payload especially on large servers with a lot of CPUs.

### Motivation

https://datadoghq.atlassian.net/browse/PROCS-3633

### Possible Drawbacks / Trade-offs

- Less available information about the CPUs
- Will have to find other way to optimize the payload if the removed information is needed for future feature

### Describe how to test/QA your changes

- run `invoke test --targets=./pkg/process/` to make sure tests are still passing as expected
- test with agent VM:
  
    - Once agent is running,  run the manual check commands to see the results of the check and confirm that the fields are not being populated (eg. only the "cores" field is a part of the cpus). 
        - `/opt/datadog-agent/embedded/bin/process-agent check process --json`
        - `/opt/datadog-agent/embedded/bin/process-agent check process_discovery --json`

Screenshot 2024-01-19 at 3 34 18 PM

- Verify Live Processes and normalization related to CPU appears correctly:

Screenshot 2024-01-22 at 2 01 10 PM
Screenshot 2024-01-22 at 2 00 52 PM

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by robertjli — APPROVED** [review]

**Review by wiyu — APPROVED** [review]
This looks great! Mind adding some tests for this.

File: pkg/process/checks/system_info_darwin_test.go [hunk 1]
```diff
@@ -36,3 +37,19 @@ func TestPatchCPUInfo(t *testing.T) {
...
+		// Checks if only the cores does not have the default value
+		assert.NotEqual(t, int32(0), cpuData.Cores)
+		assert.Equal(t, int32(0), cpuData.Number)
+		assert.Equal(t, "", cpuData.Vendor)
+		assert.Equal(t, "", cpuData.Family)
+		assert.Equal(t, "", cpuData.Model)
+		assert.Equal(t, "", cpuData.PhysicalId)
+		assert.Equal(t, "", cpuData.CoreId)
+		assert.Equal(t, int64(0), cpuData.Mhz)
+		assert.Equal(t, int32(0), cpuData.CacheSize)
```
Comment (wiyu) [inline pkg/process/checks/system_info_darwin_test.go:53]: You can use [assert.Empty()](https://pkg.go.dev/github.com/stretchr/testify/assert#Empty.

> Empty asserts that the specified object is empty. I.e. nil, "", false, 0 or either a slice or a channel with len == 0.

File: pkg/process/checks/system_info_darwin_test.go [hunk 2]
```diff
@@ -36,3 +37,19 @@ func TestPatchCPUInfo(t *testing.T) {
 		assert.Equal(t, int32(2), c.Cores)
 	}
 }
+
+func TestOnlyCorePopulatedDarwin(t *testing.T) {
+	sysInfo, _ := CollectSystemInfo()
+	for _, cpuData := range sysInfo.Cpus {
+		// Checks if only the cores does not have the default value
+		assert.NotEqual(t, int32(0), cpuData.Cores)
```
Comment (wiyu) [inline pkg/process/checks/system_info_darwin_test.go:45]: You should use [assert.Greater()](https://pkg.go.dev/github.com/stretchr/testify/assert#Greater) here. It shouldn't be negative, but this is expresses the intent more clearly.

File: pkg/process/checks/system_info_test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+package checks
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var _ statsProvider = &mockStatsProvider{}
```
Comment (wiyu) [inline pkg/process/checks/system_info_test.go:16]: This doesn't looks unused.

File: pkg/process/checks/system_info_windows_test.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var _ statsProvider = &mockStatsProvider{}
+
+type mockStatsProvider struct{}
```
Comment (wiyu) [inline pkg/process/checks/system_info_windows_test.go:18]: This is also unused.

File: .vscode/launch.json.template [hunk 5]
```diff
@@ -35,5 +35,35 @@
...
+            "type": "go",
+            "request": "launch",
+            "mode": "exec",
+            "program": "${workspaceRoot}/bin/process-agent/process-agent",
+            "env": {},
+            "args": [
+                "check",
+                "process_discovery"
+            ]
+        }
```
Comment (wiyu) [inline .vscode/launch.json.template:67]: I'm not too familiar with vscode. I think we should keep this out of the change set as it's very IDE specific. If we want this to go in we should isolate it into its own PR and provide documentation on how its used.

**Review by wiyu — COMMENTED** [review]
Thanks for adding the test! Minor comments around the assert functions to use on your tests and the vs code changes. Should be good to go after those are addressed.

**Review by daniel-taf — APPROVED** [review]

**Review by robertjli — APPROVED** [review]
Nice work!

**Pull request merged**